src/functions/index.ts

```diff
@@ -74,12 +75,38 @@ class FunctionRegistry extends Registry<FunctionDescription> {
     }
     const descr = addMetaInfoFromArg(addDescr);
     validateArguments(descr.args);
-    this.mapping[name] = createComputeFunctionFromDescription(descr);
+    this.mapping[name] = addInputHandling(descr, createComputeFunctionFromDescription(descr));
     super.add(name, descr);
     return this;
   }
 }
 
+function addInputHandling(
+  descr: FunctionDescription,
+  computeFunction: ComputeFunction<Arg, Matrix<ValueAndFormat> | ValueAndFormat>
+) {
+  return function (this: EvalContext, ...args: Arg[]) {
```

Comment:
adding a name helps to identify the function when profiling the code
```suggestion
  return function computeWithInputHandling(this: EvalContext, ...args: Arg[]) {
```